internal/pkg/log/logger.go

```diff
@@ -33,57 +38,103 @@ func (l *zapLogger) AddPrefix(prefix string) Logger {
 	return clone
 }
 
+func formatMessageUsingAttributes(message string, set *attribute.Set) string {
+	for _, keyValue := range set.ToSlice() {
+		message = strings.ReplaceAll(message, "%"+string(keyValue.Key)+"%", keyValue.Value.Emit())
+	}
```

Comment:
BTW there is also `strings.Replacer`, but it is not issue here.
https://medium.com/@vikram.ingawale91/golang-strings-replace-vs-strings-replacer-a7b2d2b71593

Reply:
Yes, bu overall, there won't be so many logs that it will have a big impact (I think).

For `strings.Replacer`, you has to allocate `oldNew` slice, which also costs something.

I mentioned it mainly so you know it exists.